lib/request_overrider.js

```diff
@@ -313,184 +316,188 @@ function RequestOverrider(req, options, interceptors, remove, cb) {
 
       }
     }
+    return continueWithResponseBody(responseBody);
 
-    //  Transform the response body if it exists (it may not exist if we have `responseBuffers` instead)
-    if(responseBody) {
-      debug('transform the response body');
 
-      if (!Buffer.isBuffer(responseBody) && !isStream(responseBody)) {
-        if (typeof responseBody === 'string') {
-          responseBody = new Buffer(responseBody);
-        } else {
-          responseBody = JSON.stringify(responseBody);
+    function continueWithResponseBody(responseBody) {
```

Comment:
I think that the callback should be using the standard callback pattern: error-first, so that it can be easily composed with other I/O.

This allows stuff like:

```js
.reply(function(cb) {
  fs.readFile(file, cb);
});
```

or even better:

```js
.reply(fs.readFile.bind(fs, file));
```

Then the answer comes: what happens when there is an error happens? I propose that it's just thrown so that the test stops.